server/server.go

```diff
@@ -42,72 +55,362 @@ func HandleWebsocketConnection(serverHTTPResponse http.ResponseWriter, clientHTT
+		} // if writeMessageError != nil
+	}
+	return nil
+}
+
+func persistSignedPreKeyFromClient(websocketConnection *websocket.Conn, signedPreKeyFromClient *backendProtobuf.PreKey) error {
+	// Create a hex representation of the IdentityKey of the client
+	clientIdentityKeyHex := hex.EncodeToString(signedPreKeyFromClient.IdentityKey)
+	// Use protobuf to marshal the signed pre key into bytes so that we can store it easily
+	signedPreKeyFromClientProtobufBytes, protoMarshalErr := proto.Marshal(signedPreKeyFromClient)
+	// If there is an error while marshalling the signed pre key from the client
+	if protoMarshalErr != nil {
+		// Return the error
+		return protoMarshalErr
+	}
+	// Store the SignedPreKey from the client
+	multiUserSignedPreKeyStore[clientIdentityKeyHex] = signedPreKeyFromClientProtobufBytes
+	// Signal to the client that the SignedPreyKey has been persisted
+	writeMessageError := websocketConnection.WriteMessage(websocket.BinaryMessage, signedPreKeyFromClientProtobufBytes)
+	// If there is an error while sending a message to a client
+	if writeMessageError != nil {
+		return writeMessageError
+	} // if writeMessageError != nil
+
+	return nil
+}
+
+func handleMessageFromClient(websocketConnection *websocket.Conn, messagesFromClient []*backendProtobuf.ChatMessage) error {
+	// For each message received from client
+	for _, singleMessageFromClient := range messagesFromClient {
+		// Use protobuf to marshal the message into bytes so that we can store it easily
+		chatMessageProtobufBytes, chatMessageProtobufError := proto.Marshal(singleMessageFromClient)
+		// If there is an error while marshaling the individual message from the client
+		if chatMessageProtobufError != nil {
+			// Return the error
+			return chatMessageProtobufError
+		} // if chatMessageProtobufError != nil
+		// Convert the identityPublicKey of the intented recipient of the message into a string so that it's easier to use
+		messageReceiverString := string(singleMessageFromClient.Receiver)
+		// Use the string representation of the identityPublicKey as part of our backend storage
+		multiUserChatMessage[messageReceiverString] = append(multiUserChatMessage[messageReceiverString], chatMessageProtobufBytes)
+		// Echo back the same message we received from the client back to him so that we inform him that the message has been persisted
+		writeMessageError := websocketConnection.WriteMessage(websocket.BinaryMessage, chatMessageProtobufBytes)
+		// If there is an error while sending a message to a client
+		if writeMessageError != nil {
+			return writeMessageError
+		} // if writeMessageError != nil
+	} // for _, singleMessageFromClient
+	return nil
+} // func handleMessageFromClient
+
+func deliverRequestedPreKeyBundle(websocketConnection *websocket.Conn, requestedPreKeyBundle []byte) {
+	// Create a string representation of the publicKey associated with the user in the preKeyBundle request
+	requestedPreKeyBundleString := string(requestedPreKeyBundle)
```

Comment:
i guess you should use `hex.EncodeToString`

Reply:
In the test I wrote I sent this value already hex encoded, just because I remember Alberto was having some issues far back when dealing with sending invalid bytes over network or something similar. 

I think think since then the issue has been resolved. 

Will the clients always be sending the pure non-hexEncoded bytes over the network?